### PR TITLE
[PNUDEV-54] Simplify local development

### PR DESCRIPTION
**Pull request by pikulo-kama**

### CODE REVIEW

**Comment by Igor-Polatajko** [comment]
You should read the description of the task more carefully.
It says: "Use **stub** bean of VideoApiService for "local" profile".

Also, please, mention the `default` profile in readme, which should be used for production as well.

File: radio-service-api/src/main/java/com/pnu/dev/radioserviceapi/service/VideoApiServiceImpl.java [hunk 1]
```diff
@@ -22,6 +23,7 @@
 
 import static java.util.Objects.nonNull;
 
+@Profile("local")
```
Comment (Igor-Polatajko) [inline radio-service-api/src/main/java/com/pnu/dev/radioserviceapi/service/VideoApiServiceImpl.java:26]: Sorry, I mentioned the wrong class in the description.
I meant `VideoApiClient`.

But, anyway, we don't want the object of the real class to be loaded to the context, when we use this profile.
It have to be stub, for this profile and real object for default one.

**Comment by Igor-Polatajko** [comment]
Also, please, follow the naming convention for the PRs

File: radio-service-api/src/main/java/com/pnu/dev/radioserviceapi/service/VideoApiServiceImpl.java [hunk 2]
```diff
@@ -13,6 +13,7 @@
 import com.pnu.dev.radioserviceapi.util.OperationResult;
 import com.pnu.dev.radioserviceapi.util.mapper.VideoMapper;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Profile;
```
Comment (Igor-Polatajko) [inline radio-service-api/src/main/java/com/pnu/dev/radioserviceapi/service/VideoApiServiceImpl.java:16]: unused import

File: radio-service-api/src/main/java/com/pnu/dev/radioserviceapi/client/dto/search/YoutubeSearchResponse.java [hunk 3]
```diff
@@ -1,10 +1,11 @@
 package com.pnu.dev.radioserviceapi.client.dto.search;
 
+import lombok.Builder;
 import lombok.Data;
 
 import java.util.List;
 
-
+@Builder(toBuilder = true)
```
Comment (Igor-Polatajko) [inline radio-service-api/src/main/java/com/pnu/dev/radioserviceapi/client/dto/search/YoutubeSearchResponse.java:8]: do you use toBuilder() for all the classes, where you added it?

File: radio-service-api/src/main/java/com/pnu/dev/radioserviceapi/client/YoutubeApiClientStub.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+import java.util.stream.Collectors;
+
+@Profile({"default", "local"})
+@Service
+public class YoutubeApiClientStub implements YoutubeApiClient {
+
+    List<YoutubeVideoDetails> videos;
+
+    @AllArgsConstructor
+    private class YoutubeVideoDetails {
```
Comment (Igor-Polatajko) [inline radio-service-api/src/main/java/com/pnu/dev/radioserviceapi/client/YoutubeApiClientStub.java:25]: no need in this class

**Review by Igor-Polatajko — CHANGES_REQUESTED** [review]

File: radio-service-api/src/main/java/com/pnu/dev/radioserviceapi/client/dto/search/YoutubeSearchResponse.java (hunk 3 above)
Comment (pikulo-kama) [reply radio-service-api/src/main/java/com/pnu/dev/radioserviceapi/client/dto/search/YoutubeSearchResponse.java:8]: Any of these classes do not have constructors or setters so field initialization is impossible. That is why I decided to use _toBuilder_

File: radio-service-api/src/main/java/com/pnu/dev/radioserviceapi/client/YoutubeApiClientImpl.java [hunk 5]
```diff
@@ -7,12 +7,13 @@
...
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Profile;
 import org.springframework.stereotype.Component;
 import org.springframework.web.client.RestTemplate;
 import org.springframework.web.util.UriComponents;
 import org.springframework.web.util.UriComponentsBuilder;
 
-
+@Profile("default")
```
Comment (Igor-Polatajko) [inline radio-service-api/src/main/java/com/pnu/dev/radioserviceapi/client/YoutubeApiClientImpl.java:16]: No need in this annotation.
`default` is default

File: README.md [hunk 6]
```diff
@@ -29,6 +29,10 @@ java <env variables: -DvariableName='value'> -jar <jar location>
 - ADMIN_USERNAME (_default_: 'admin') - _admin panel username_
 - ADMIN_PASSWORD (_default_: 'admin') - _admin panel password_
 
+## Spring profiles
+- default - *profile used in production(uses real Youtube client and requires YOUTUBE_API_KEY)*
```
Comment (Igor-Polatajko) [inline README.md:33]: Space, before "(".
Comma, after profile

File: radio-service-api/src/main/java/com/pnu/dev/radioserviceapi/client/YoutubeApiClientStub.java [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+@Profile("local")
+@Service
+public class YoutubeApiClientStub implements YoutubeApiClient {
+
+    @Override
+    public OperationResult<YoutubeSearchResponse> findRecentVideos() {
+
+        YoutubeSearchResponse response = new YoutubeSearchResponse(){{
+            setItems(Arrays.asList(
+                    new ItemYoutubeSearchResponse(){{
```
Comment (Igor-Polatajko) [inline radio-service-api/src/main/java/com/pnu/dev/radioserviceapi/client/YoutubeApiClientStub.java:24]: Consider using builders everywhere

**Review by Igor-Polatajko — APPROVED** [review]

**Pull request merged**